### PR TITLE
[GH-5951] Updates Authentication > SAML 2.0 Settings to New Format

### PR DESCRIPTION
**Pull request by matthew-src**
#### Summary

This PR updates the SAML 2.0 section of the Authentication settings to the new format. The description of these settings should get a technical review:

1. Ignore guest users when synchronizing with AD/LDAP
2. Override SAML bind data with AD/LDAP information
3. Enable encryption
4. Sign request - The default for this setting was not clear. Also, the current docs indicate it take a string value, but I think it is actually a boolean value.
5. Enable admin and Admin attribute - The functioning of these two settings is not clear to me.

#### Ticket Link

This PR partially addresses https://github.com/mattermost/docs/issues/5951.

### CODE REVIEW

**Comment by cwarnermm** [comment]
Thanks, @matthew-src! I've generated a preview of these updates. In that preview, there are formatting syntax issues with the following sections:
- Override SAML bind data with AD/LDAP information (note text bullets)
- Verify signature (table)
- Service provider login URL (table)
- Sign request (table)
- Enable admin attribute (table)
- Login button text (table)

@mkraft - Would a member of the Suite Users team be available to technically review the SAML 2.0 config setting documentation for accuracy, please?

**Comment by matthew-src** [comment]
>     * Override SAML bind data with AD/LDAP information (note text bullets)
> 
>     * Verify signature (table)
> 
>     * Service provider login URL (table)
> 
>     * Sign request (table)
> 
>     * Enable admin attribute (table)
> 
>     * Login button text (table)

I am sorry about that. I think the tables are all fixed now. I am not sure about the note text bullets. I think they just needed an extra space for indentation.

**Comment by matthew-src** [comment]
I reviewed this again. **Now** the Override table is fixed.

**Comment by cwarnermm** [comment]
Not a problem at all, @matthew-src! RST is a prescriptive file format and it can take a couple of runs to get the syntax working as expected. Generated previews help this process a lot!

You're doing such a great job with this project, and we're so incredibly thankful for your efforts.

**Review by cwarnermm — APPROVED** [review]
Thanks, @matthew-src! I'll hold off merging this one until Engineering's had an opportunity to review.

**Review by mkraft — CHANGES_REQUESTED** [review]

File: source/configure/authentication-configuration-settings.rst [hunk 1]
```diff
@@ -724,15 +720,11 @@ Ignore guest users when synchronizing with AD/LDAP
...
-+------------------------------------------------------------------------------------------------------------------+
++-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------+
+| - **true**: When syncing with the AD/LDAP server, Mattermost ignores Guest Users that have been deactivated and removed, and does not make any changes to them. Manage guest deactivation manually via **System Console > Users**. For more information, see `AD/LDAP Setup <https://docs.mattermost.com/onboard/ad-ldap.html>`__ and `Advanced User Management <https://docs.mattermost.com/guides/administration.html#advanced-user-management>`__. | - System Config path: **Authentication > SAML 2.0**                      |
```
[the remark this answers is not in the document]
Comment (matthew-src) [reply source/configure/authentication-configuration-settings.rst:724]: Thank you, @mkraft, for reviewing this PR. If I understand correctly, I would try to summarize the functionality as:

When "Ignore guest users..." is set to **true**, syncing SAML-authenticated Mattermost accounts with AD/LDAP has no effect on Mattermost guest accounts that were created based on a Guest User AD/LDAP filter (see AD/LDAP filter setting). Changes to the user role or account deactivation must be done manually.

Now, just to be clear, would role changes be synced if "Ignore guest users..." is set to **false**?

**Review by mkraft — APPROVED** [review]

**Comment by cwarnermm** [comment]
@matthew-src - Is there anything outstanding left on this PR from your perspective?

File: source/configure/authentication-configuration-settings.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkraft) [reply source/configure/authentication-configuration-settings.rst:724]: @matthew-src I realize now the description I put above was all about the AD/LDAP guest filters but I put it as a comment on the documentation for SAML ignore-guest boolean configuration.

Everything I said above is true for the AD/LDAP filter.

There's also a guest attribute for SAML which is documented correctly in product:

* changes to the guest attribute won't promote a user from guest to regular user
* changes to the guest filter *can* demote a regular user to a guest

If `IgnoreGuestsLdapSync` setting is `true` a user authenticating via SAML doesn't even need to exist in AD/LDAP and no values (ex. email, firstname, lastname) will be update in Mattermost for that user from AD/LDAP.

**Comment by matthew-src** [comment]
I made a few changes, and some formatting fixes, but it all set now.

File: source/configure/authentication-configuration-settings.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthew-src) [reply source/configure/authentication-configuration-settings.rst]: Thank you, @mkraft . I will review the AD/LDAP settings based on your comment.

**Pull request merged**